app/routes/_gcn.circulars._archive._index/route.tsx

```diff
@@ -115,24 +118,27 @@ export async function action({ request }: ActionFunctionArgs) {
         },
         user
       )
-      result = await get(parseFloat(circularId))
+      newItem = await get(parseFloat(circularId))
       break
     case 'new':
-      result = await put({ ...props, submittedHow: 'web' }, user)
+      newItem = await put({ ...props, submittedHow: 'web' }, user)
       break
     default:
       break
   }
-  return result
+  return { newItem, showSuccess }
```

Comment:
Rather than call it `result`, how about `newCircular`?

Reply:
In the action? Or in the component function? In the latter that will lead to it looking like 

```ts
const newCircular= useActionData<typeof action>() 
// const {newItem, intent} = useActionData<typeof action>() throws 
// errors about newItem and intent not being defined
...
const allItems = [
    ...(newCircular?.newItem ? [newCircular.newItem] : []),
    ...(items || []),
  ]
```

Reply:
```suggestion
  return { newCircular, intent }
```